### ISSUE TITLE
Provide a server implementation of the LocalStateQuery protocol

### ISSUE THREAD

**Issue by mrBliss** [input-output-hk/ouroboros-network#1366]
The wallet will communicate with the node using the `LocalStateQuery` protocol: https://github.com/input-output-hk/ouroboros-network/blob/2c4fa419346ab2c24e12ba7950fde4faed173191/ouroboros-network/src/Ouroboros/Network/Protocol/LocalStateQuery/Type.hs

Consensus must implement the server side of this protocol. It should be relatively straightforward to implement it using the LedgerDB.

**Comment by mrBliss** [comment]
This will only be needed for Shelley, apparently.

**Comment by dcoutts** [comment]
@KtorZ now says that he'd like to use this for Byron too, since it gives him a more consistent design across Byron & Shelley.

I think the set of local state queries available will need to be part of one of the protocol classes, since they're specific to the ledger types.

Note also this ticket depends on #1412 which is the remaining network layer work for the protocol.

**Comment by dcoutts** [comment]
The queries are obviously specific to the ledger type.

The one Matthias needs for Byron is to grab the whole update system state (which includes the current protocol params and pending (confirmed) update proposals). (We may wish to explicitly exclude unconfirmed updates and votes)

**Comment by KtorZ** [comment]
I am not sure where to log this so, for lack of a better place I'll put it here: 

For Byron, @dcoutts has already captured the requirements / needed pieces of the ledger state (updatable protocol params & confirmed update proposals). 

For Shelley, we are looking for:

- All the Byron's requirements
- The rewards given to a particular account. The latest balance would be a must-have, while the complete history would be a nice-to-have (although I am not sure of the feasibility for the node. This is something we would likely keep in an explorer). 
- The stake distribution of past epochs, per epoch, (up to a certain age, last `k` is nice but also much more than what we would need) in terms of pools ids (who owns what?).
- The total reward amount distributed in past epochs, per epoch (i.e. the `R` in the incentive paper). 

Cheers!
